neutrons/reflectivity_ui
# Add plot toolbar buttons for log x-axis, log y-axis and R*Q^4

Resolves [Story 3205: [QUICKNXS] Add button to plot R(Q)*Q^4](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=3205)

# Short description of the changes:
- Refactor class `NavigationToolbar` and create child classes `NavigationToolbarGeneric` and `NavigationToolbarReflectivity`
- Add three new buttons to  `NavigationToolbarReflectivity` to:
    - toggle linear/log x-axis
    - toggle linear/log y-axis
    - toggle between plotting `R` and `R*Q^4`
- Add missing icons to existing navigation toolbar actions "Lines" and "Save data"
- Add unit tests

# Long description of the changes:
The button to toggle `R*Q^4` only manipulates the data points stored in the `matplotlib` plot objects. The underlying data used to generate the plots are left unchanged.

# Check list for the pull request

# Check list for the reviewer

# Manual test for the reviewer
Open QuickNXS and test the different navigation toolbar buttons. There are now two different navigation toolbars, one generic and one for reflectivity plots. See the EWM story for example of data to load.

![Screenshot from 2024-01-16 08-54-26 - annotated](https://github.com/neutrons/reflectivity_ui/assets/6989921/1efd6643-2b13-4336-bca2-14a267248839)

# References
 [Story 3205: [QUICKNXS] Add button to plot R(Q)*Q^4](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=3205)
